Nure/Update/NuGetWrapper.cs

```diff
@@ -0,0 +1,157 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using NLog;
+using Nure.Configuration;
+
+namespace Nure.Update
+{
+    public class NuGetWrapper
+    {
+        private const string NUGET_PACKAGE_HEADER = "Top-level Package";
+        private const string DOTNET_PROJECT_SEPARATOR = "----------";
+
+        private static readonly ILogger s_Logger = LogManager.GetCurrentClassLogger();
+        private static readonly Regex s_ProjectNameRegex = new Regex("(?<=Project `)(.*?)(?=` has the following updates to its packages)", RegexOptions.Compiled);
+
+        private readonly NureOptions m_NureOptions;
+        private readonly string m_RepositoryDirectory;
+        private readonly List<string> m_MonitoredProjects;
+
+        public NuGetWrapper(NureOptions p_NureOptions,
+                            string p_RepositoryDirectory)
+        {
+            m_NureOptions = p_NureOptions;
+            m_RepositoryDirectory = p_RepositoryDirectory;
+            m_MonitoredProjects = GetProjects();
+        }
+
+        public void Run()
+        {
+            Process process = GetPackageListProcess();
+            process.Start();
+            Dictionary<string, List<NuGetPackage>> packages = GetUpdatesFromStream(process.StandardOutput);
+            s_Logger.Info($"Found {packages.Count} projects with packages to update.");
+            UpdatePackages(packages);
+            s_Logger.Info("Finished updating dependencies.");
+        }
+
+        private Dictionary<string, List<NuGetPackage>> GetUpdatesFromStream(StreamReader p_InputStream)
+        {
+            Dictionary<string, List<NuGetPackage>> packagesByProject = new Dictionary<string, List<NuGetPackage>>();
+
+            bool isReadingPackages = false;
+            string currentProject = "";
+            string currentLine;
+
+            while ((currentLine = p_InputStream.ReadLine()) != null) {
+                if (string.IsNullOrWhiteSpace(currentLine)) {
+                    continue;
+                }
+
+                if (currentLine.Contains(NUGET_PACKAGE_HEADER)) {
+                    isReadingPackages = true;
+                    continue;
+                }
+
+                Match regexMatch = s_ProjectNameRegex.Match(currentLine);
+
+                if (regexMatch.Success) {
+                    isReadingPackages = false;
+
+                    currentProject = m_MonitoredProjects.First(project => project.Contains(regexMatch.Value, StringComparison.InvariantCultureIgnoreCase));
+                    packagesByProject.Add(currentProject, new List<NuGetPackage>());
+                } else if (isReadingPackages) {
+                    List<string> packageComponents = currentLine.Split(" ", StringSplitOptions.RemoveEmptyEntries).ToList();
+
+                    if (packageComponents.Count == 5) {
```

Comment:
Wuuuut
do you have a line example? I take it the split will always return 5?

Reply:
![image](https://user-images.githubusercontent.com/13049746/97471310-f7151780-191e-11eb-8e8d-e00f7e1feea2.png)